modules/dkimVerifier.jsm.js

```diff
@@ -353,10 +359,26 @@ var Verifier = (function() {
 	 * @return {Boolean}
 	 */
 	function verifyED25519Sig(key, str, hash_algo, signature, warnings, _keyInfo = {}) {
+		let result = false;
 		let hashedStr = dkim_hash(str, hash_algo, "b64");
-		return NaCl.nacl.sign.detached.verify(NaClUtil.nacl.util.decodeBase64(hashedStr),
-											NaClUtil.nacl.util.decodeBase64(signature),
-											NaClUtil.nacl.util.decodeBase64(key));
+		let hashedStr_byte = ED25519.nacl.util.decodeBase64(hashedStr);
+		let signature_byte = ED25519.nacl.util.decodeBase64(signature);
+		let key_byte = ED25519.nacl.util.decodeBase64(key);
+		// each byte has 8 bit (a valid key_byte array has a length of 32)
+		let keyLength = key_byte.length * 8;
+		log.debug("ED25519 key length: " + keyLength);
+		if (hash_algo !== "sha256") {
+			throw new DKIM_SigError("DKIM_SIGERROR_KEY_HASHNOTINCLUDED");
+		}
+		if (keyLength !== 256) {
+			throw new DKIM_SigError("DKIM_SIGERROR_KEYDECODE");
+		}
```

Comment:
Not against adding this checks, but would be interested to know if there was a special reason you added the key length calculation, e.g. you actually encountered a key that failed this.

Reply:
No, I didn’t encounter a wrong key in real life.
As I wrote in the issue about displaying more details on dkim signatures, I‘d also like to add the key length to algorithm details (I have regularly the weak key warning for RSA in real life and it would be nice to see the key length on a glance).
Therefore, I had a deeper look into RSA verification method and wanted the ED25519 verification similar. 
Instead of returning a 256bit key length for the details every time, I had a look on how to get the „real“ key length and also added error handling for „wrong hash algorithm“ and „wrong key length“.